src/du_parameters/carrier_bandwidth.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Calculate Carrier Bandwidth using bandwidth and subcarrier spacing."""
+
+import decimal
+import logging
+from decimal import Decimal
+
+from src.du_parameters.frequency import (
+    Frequency,
+)
+from src.du_parameters.guard_band import GuardBandError, get_minimum_guard_band
+
+logger = logging.getLogger(__name__)
+
+CONFIG_CONSTANT_TWO = Decimal("2")
+NUMBER_OF_SUBCARRIERS = Decimal("12")
+
+
+class CarrierBandwidthError(Exception):
+    """Exception raised when carrier bandwidth calculation fails."""
+
+    pass
+
+
+def get_carrier_bandwidth(bandwidth: Frequency, subcarrier_spacing: Frequency) -> int:
+    """Calculate channel bandwidths of the carrier.
+
+    Result should be in terms of Resource Blocks (RB) using bandwidth and subcarrier spacing.
+
+    Args:
+        bandwidth (Frequency): bandwidth
+        subcarrier_spacing (Frequency): subcarrier spacing
+
+    Returns:
+        carrier bandwidth (int): The channel bandwidths of the carrier
+         in terms of Resource Blocks (RB)
+
+    Raises:
+        TypeError: If bandwidth or subcarrier spacing are not of type Frequency.
+        ValueError: If bandwidth or subcarrier spacing are not greater than 0.
+        CarrierBandwidthError: If calculation fails or Guard band calculation fails.
+    """
+    if not isinstance(bandwidth, Frequency) or not isinstance(subcarrier_spacing, Frequency):
+        logger.error("Both bandwidth and subcarrier_spacing must be of type Frequency.")
+        raise TypeError("Both bandwidth and subcarrier_spacing must be of type Frequency.")
+
```

Comment:
```suggestion
```
There are already type hints and the code will already fail if the wrong types are passed in anyway.